tensorzero-core/src/error.rs

```diff
@@ -137,6 +137,10 @@ impl Error {
     pub fn log(&self) {
         self.0.log();
     }
+
+    pub fn nonretryable(&self) -> bool {
```

Comment:
nit: either `is_nonretryable` or maybe `is_retryable` so avoid double-negatives later and we can write `retry().when(e.is_retryable())`?

Reply:
makes sense, done